pkg/oc/clusteradd/components/openshift-logging/openshift-logging.go

```diff
@@ -0,0 +1,99 @@
+package openshift_logging
+
+import (
+	"encoding/base64"
+	"fmt"
+	"github.com/golang/glog"
+	"io/ioutil"
+	"net/url"
+	"path"
+
+	configapi "github.com/openshift/origin/pkg/cmd/server/apis/config"
+	configapilatest "github.com/openshift/origin/pkg/cmd/server/apis/config/latest"
+	kerrs "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/openshift/origin/pkg/oc/clusteradd/componentinstall"
+	"github.com/openshift/origin/pkg/oc/clusterup/coreinstall/kubeapiserver"
+	"github.com/openshift/origin/pkg/oc/clusterup/docker/dockerhelper"
+	"github.com/openshift/origin/pkg/oc/clusterup/manifests"
+	"github.com/openshift/origin/pkg/oc/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	olNamespace = "openshift-logging"
+)
+
+type OpenshiftLoggingComponentOptions struct {
+	InstallContext componentinstall.Context
+}
+
+func (c *OpenshiftLoggingComponentOptions) Name() string {
+	return "openshift-logging"
+}
+
+func (c *OpenshiftLoggingComponentOptions) Install(dockerClient dockerhelper.Interface) error {
+	adminKubeConfigBytes, err := ioutil.ReadFile(path.Join(c.InstallContext.BaseDir(), kubeapiserver.KubeAPIServerDirName, "admin.kubeconfig"))
+	adminKubeConfigString := base64.StdEncoding.EncodeToString(adminKubeConfigBytes)
+
+	configBytes, err := ioutil.ReadFile(path.Join(c.InstallContext.BaseDir(), kubeapiserver.KubeAPIServerDirName, "master-config.yaml"))
+	if err != nil {
+		return err
+	}
+	configObj, err := runtime.Decode(configapilatest.Codec, configBytes)
+	if err != nil {
+		return err
+	}
+	masterConfig, ok := configObj.(*configapi.MasterConfig)
+	if !ok {
+		return fmt.Errorf("the %#v is not MasterConfig", configObj)
+	}
+	masterUrl, err := url.Parse(masterConfig.MasterPublicURL)
+	if err != nil {
+		return err
+	}
+
+	kubeAdminClient, err := kubernetes.NewForConfig(c.InstallContext.ClusterAdminClientConfig())
+	if err != nil {
+		return errors.NewError("cannot obtain API clients").WithCause(err)
+	}
+
+	params := map[string]string{
+		"NAMESPACE":        olNamespace,
+		"PUBLIC_HOSTNAME":  masterUrl.Hostname(),
+		"ADMIN_KUBECONFIG": adminKubeConfigString,
+	}
+	glog.V(2).Infof("instantiating openshift logging template with parameters %v", params)
+
+	err = kubeAdminClient.BatchV1().Jobs(olNamespace).Delete("openshift-logging-apb", metav1.NewDeleteOptions(0))
```

Comment:
Your component add logic needs to be indempotent.  I don't see an obvious check here.  Is the "am I already ready?" check done in this job?

Reply:
The openshift-ansible logging roles in the container run by the job are idempotent. After rerunning logging continues to function correctly.